newsroom/agenda/service.py

```diff
@@ -0,0 +1,118 @@
+from datetime import datetime
+
+from newsroom.auth.utils import get_user_from_request, get_company_from_request
+from newsroom.agenda.agenda import (
+    is_events_only_access,
+    build_agenda_query,
+    nested_query,
+    planning_items_query_string,
+    aggregations,
+    remove_fields,
+    PLANNING_ITEMS_FIELDS,
+)
+from newsroom.agenda.model import FeaturedResourceModel
+from newsroom.utils import get_local_date
+from newsroom.template_filters import is_admin
+
+from superdesk.flask import abort
+from superdesk.core.resources import AsyncResourceService
+from superdesk.utc import local_to_utc
+from superdesk.utils import ListCursor
+
+
+class FeaturedService(AsyncResourceService[FeaturedResourceModel]):
+    resource_name = "agenda_featured"
+
+    async def on_create(self, docs):
+        """
+        Add UTC from/to datetimes on save.
+        Problem is 31.8. in Sydney is from 30.8. 14:00 UTC to 31.8. 13:59 UTC.
+        And because we query later using UTC, we store those UTC datetimes as
+        display_from and display_to.
+        """
+        for item in docs:
+            date = datetime.strptime(item._id, "%Y%m%d")
+            item.display_from = local_to_utc(item.tz, date.replace(hour=0, minute=0, second=0))
+            item.display_to = local_to_utc(item.tz, date.replace(hour=23, minute=59, second=59))
+        await super().on_create(docs)
+
+    async def find_one_for_date(self, for_date: datetime) -> FeaturedResourceModel | None:
+        return await self.find_one(req=None, display_from={"$lte": for_date}, display_to={"$gte": for_date})
+
+    async def get_featured_stories(self, req, lookup):
```

Comment:
Instead of `req, lookup` (which is the old way of doing things), you can directly provide the arguments you require here, and then use `date_from` and `timezone_offset` directly. Such as:
```python
async def get_featured_stories(self, date_from: str, timezone_offset: int = 0) -> Any:
```

I wouldn't worry too much about the return type, this will change anyhow when the Agenda resource itself is upgraded, so leaving it as `Any` should suffice for now